src/Microsoft.DotNet.Darc/src/Darc/Operations/CloneOperation.cs

```diff
@@ -273,7 +273,7 @@ private static async Task HandleMasterCopyWithDefaultGitDir(RemoteFactory remote
             {
                 log.LogInformation($"Cloning master copy of {repoUrl} into {masterGitRepoPath}");
                 IRemote repoRemote = await remoteFactory.GetRemoteAsync(repoUrl, log);
-                repoRemote.Clone(repoUrl, null, masterGitRepoPath, masterRepoGitDirPath);
+                repoRemote.Clone(repoUrl, null, masterGitRepoPath, checkoutSubmodules: true, masterRepoGitDirPath);
```

Comment:
Instead of having to rewrite all these calls (unless some are actually going to become checkoutSubmodules: false), can we just change the existing function to call the new one with checkoutSubmodules: true? So like:

```
public void Clone(string repoUri, string commit, string targetDirectory, string gitDirectory = null)
{
    Clone(repoUri, commit, targetDirectory, true, _logger, _personalAccessToken, gitDirectory);
}

public void Clone(string repoUri, string commit, string targetDirectory, bool checkoutSubmodules, string gitDirectory = null)
{
    Clone(repoUri, commit, targetDirectory, checkoutSubmodules, _logger, _personalAccessToken, gitDirectory);
}
```

Reply:
Should we rather have `CloneWithoutSubmodules`?

Reply:
I'm happy with the overload of Clone, honestly. I just think it's easier if the next person to come along doesn't need to think "do I need submodules or not?" when they just want to just some default clone behavior. Up to you. This was more of a nit than anything else

Reply:
> unless some are actually going to become `checkoutSubmodules: false`

The goal is for some to become `false` - that's the purpose of this change.

> when they just want to just some default clone behavior

I understand but then regular git clone doesn't do that so it would make sense to default on `false` actually?

Reply:
Thinking about this more - since the default behaviour of the API is currently not equal to the default behaviour of `git clone`, I think it might be for the best to always make the user be explicit about the submodules since calling the clone API only will actually not call clone only but also initialize submodules.

There are only 2 calls I had to change so it's not too many and this way we also stay backwards compatible (which is good because I don't have visibility into who uses this library outside of arcade-services).